.hooks/pre-commit

```diff
@@ -60,6 +60,16 @@ ui_lint() {
   $LINTER || block "UI lint failed"
 }
 
+backend_lint() {
+  # Silently succeed if no changes staged for Go code folders.
+  DIRS="api/ audit/ builtin/ command/ http/ internal/ internalshared/ pkg/ plugins/ sdk/ shamir/ vault/"
+  if git diff --name-only --cached --exit-code -- $DIRS; then
```

Comment:
Instead of worrying about a list of folders, why not just limit ourselves to `*.go` files?

Reply:
Ah, we could do `git diff --name-only --cached --exit-code -- '*.go'` to explicitly avoid globbing, I like it.